### PR TITLE
Created Question Controller

### PR DESCRIPTION
**Pull request by swapdutt**
swapnadeep.dutta: 
1. Added QuestionEntity.java and AnswerEntity.java
2. Added QuestionDao.java
3. Added QuestionBusinessService.java
4. Added QuestionController.java
5. Added test cases in QuestionControllerTest.java
6. Added QuestionEntity dependency in UserEntity.java

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * The method is used to delete the question posted by user
+     *
+     * @param questionEntity
+     * @return
+     */
+
+    public QuestionEntity deleteQuestion(QuestionEntity questionEntity) {
+        // TODO Auto-generated method stub
```
Comment (cushivas) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java:82]: We should remove TODO's

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * The method is used to update the question posted by user
+     *
+     * @param questionEntity
+     * @return
+     */
+
+    public QuestionEntity updateQuestion(QuestionEntity questionEntity) {
+        // TODO Auto-generated method stub
```
Comment (cushivas) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java:69]: We should remove TODO's

**Review by cushivas — APPROVED** [review]
Minor things inside, we will fix it in next commit

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java (hunk 1 above)
Comment (swapdutt) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java:82]: These are auto comments whenever we create a method. This won't affect the project

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java (hunk 2 above)
Comment (swapdutt) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java:69]: These are auto comments whenever we create a method. This won't affect the project

**Review by cushivas — APPROVED** [review]
Looks good.

**Pull request merged**